plotly_resampler/aggregation/aggregation_interface.py

```diff
@@ -4,169 +4,260 @@
 
 import re
 from abc import ABC, abstractmethod
-from typing import List, Tuple
+from typing import List, Optional, Tuple
 
 import numpy as np
-import pandas as pd
 
 
-class AbstractSeriesAggregator(ABC):
-    """"""
-
+class AbstractAggregator(ABC):
     def __init__(
         self,
         interleave_gaps: bool = True,
-        nan_position: str = "end",
-        dtype_regex_list: List[str] = None,
+        x_dtype_regex_list: Optional[List[str]] = None,
+        y_dtype_regex_list: Optional[List[str]] = None,
     ):
         """Constructor of AbstractSeriesAggregator.
 
         Parameters
         ----------
         interleave_gaps: bool, optional
             Whether None values should be added when there are gaps / irregularly
-            sampled data. A quantile-based approach is used to determine the gaps /
+            sampled data. An x-range based approach is used to determine the gaps /
             irregularly sampled data. By default, True.
-        nan_position: str, optional
-            Indicates where nans must be placed when gaps are detected. \n
-            If ``'end'``, the first point after a gap will be replaced with a
-            nan-value \n
-            If ``'begin'``, the last point before a gap will be replaced with a
-            nan-value \n
-            If ``'both'``, both the encompassing gap datapoints are replaced with
-            nan-values \n
-            .. note::
-                This parameter only has an effect when ``interleave_gaps`` is set
-                to *True*.
-        dtype_regex_list: List[str], optional
-            List containing the regex matching the supported datatypes, by default None.
+        x_dtype_regex_list: List[str], optional
+            List containing the regex matching the supported datatypes for the x array,
+            by default None.
+        y_dtype_regex_list: List[str], optional
+            List containing the regex matching the supported datatypes for the y array,
+            by default None.
 
         """
         self.interleave_gaps = interleave_gaps
-        self.dtype_regex_list = dtype_regex_list
-        self.nan_position = nan_position.lower()
-        super().__init__()
+        self.x_dtype_regex_list = x_dtype_regex_list
+        self.y_dtype_regex_list = y_dtype_regex_list
 
-    @abstractmethod
-    def _aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
-        pass
+    @staticmethod
+    def _calc_med_diff(x_agg: np.ndarray) -> Tuple[float, np.ndarray]:
+        # ----- divide and conquer heuristic to calculate the median diff ------
```

Comment:
Can we make this a docstring instead of the # Comment? Even if internal, a single-quote underneath would help it show up in the IDE better..

Reply:
DONE!